client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -1187,6 +1148,94 @@ public synchronized boolean reassignShuffleServers(
     }
   }
 
+  @Override
+  public ShuffleServerInfo reassignFaultyShuffleServer(
```

Comment:
Why not using `synchronized` , if having multiple request happen at the same time? 

And if the same shuffleServerID is faulty from different tasks, we should make it use the same assignment rule from the first retry assign here?

Reply:
Got it.